roles/common/tasks/networking.yml

```diff
@@ -13,6 +13,36 @@
   with_items: network_interfaces_d
   tags: ['prereboot']
 
+- name: set gateway address for multi subnet 
+  set_fact: dfl_nw_gw="{{ item.gateway }}"
+  when: (update_routes|default('false')|bool) and ((ansible_default_ipv4.network ==  ( item.cidr | ipaddr('network') ))) 
+  with_items: "{{ undercloud_cidr }}"
```

Comment:
although, do we ever expect there to be more than one subnet with a default gateway?  Here's why. `set_fact` and `with_items` works in a way that essentially leaves the last item as the set fact. You'll only ever get one value out of this loop, and it'll be the value of the last thing that passed the conditional.

Is that expected?